src/pipeline/src/etl/processor.rs

```diff
@@ -149,16 +156,47 @@ impl TryFrom<&Vec<yaml_rust::Yaml>> for Processors {
 
     fn try_from(vec: &Vec<yaml_rust::Yaml>) -> Result<Self, Self::Error> {
         let mut processors = vec![];
-
+        let mut all_output_keys = HashSet::with_capacity(50);
+        let mut all_required_keys = HashSet::with_capacity(50);
+        let mut all_required_original_keys = HashSet::with_capacity(50);
         for doc in vec {
-            processors.push(parse_processor(doc)?);
+            let processor = parse_processor(doc)?;
+
+            // get all required keys
+            let processor_required_keys: Vec<String> = processor
+                .fields()
+                .iter()
+                .map(|f| f.input_field.name.clone())
+                .collect();
+
+            for key in &processor_required_keys {
+                if !all_output_keys.contains(key) {
+                    all_required_original_keys.insert(key.clone());
+                }
+            }
+
+            all_required_keys.extend(processor_required_keys);
+
+            let processor_output_keys = processor.output_keys().into_iter();
```

Comment:
Do we need to update `all_out_keys` before line 174? Which is checking the key is not in `all_output_keys` before adding it to `all_required_original_keys`.

Reply:
There is no need, we only care about the processor `all_out_keys` before the current processor.